vars/buildPlugin.groovy

```diff
@@ -152,28 +140,79 @@ def call(Map params = [:]) {
                         }
 
                         stage("Archive (${stageIdentifier})") {
-                            if (isMaven && archiveFindbugs) {
-                                def fp = [pattern: params?.findbugs?.pattern ?: '**/target/findbugsXml.xml']
-                                if (params?.findbugs?.unstableNewAll) {
-                                    fp['unstableNewAll'] = "${params.findbugs.unstableNewAll}"
+                            if (failFast && currentBuild.result == 'UNSTABLE') {
+                                error 'There were test failures; halting early'
+                            }
+
+                            if (first) {
+                                referenceJobName = env.JOB_NAME.substring(0, env.JOB_NAME.lastIndexOf("/") + 1) + "master"
+                                echo "Recording static analysis results on '${stageIdentifier}' using reference job '${referenceJobName}'"
+
+                                recordIssues enabledForFailure: true,
+                                        tool: mavenConsole(),
+                                        trendChartType: 'TOOLS_ONLY',
+                                        referenceJobName: referenceJobName
+                                recordIssues enabledForFailure: true,
+                                        tools: [java(), javaDoc()],
```

Comment:
With the addition of Checks, we are suddenly getting tons and tons of warnings. For example, https://github.com/jenkinsci/ldap-plugin/pull/48/files is practically illegible because of irrelevant, low-priority warnings like

```
NORMAL:
no @param for rootDN
```

Worse, `checks-api` apparently submits annotations even for lines of code which are not modified by a PR and GH displays every annotation, so the **Files** tab is lost in annotations which are not even related to the diff.

Can we at least turn off display of low-priority Javadoc warnings?

Reply:
Hmm, do not seem to be getting these Checks results in PRs on other plugins. Not sure what would be different here.

Reply:
You could work around this by doing one of:

* fixing the javadoc issues
* adjusting the maven javadoc plugin not to complain about these
* extending the javadoc / java parts of buildPlugin to allow overriding the configuration like you can do for spotbugs, checkstyle, pmd already and then do something like `buildPlugin(javadoc: skipPublishingChecks: true`) in your plugin
* turning checks off by default for javadoc

thoughts?

cc @XiongKezhi

Reply:
> adjusting the maven javadoc plugin not to complain about these

Could be done in `plugin-pom` I guess. I care about genuine problems—`@link` target missing, mistyped tag, etc.—but not just silliness like missing `@param` which is true for _most_ methods in the Jenkins source base.

Reply:
Those warnings are also reported by the warnings-ng plugin: https://ci.jenkins.io/job/Plugins/job/ldap-plugin/job/PR-48/3/, they are just scanned from Java or JavaDoc log files,  we can't really do much about the scan and collect process. But you can control the minimal severity of issues to report:
```
recordIssues tool: java(pattern: '*.log'), healthy: 10, unhealthy: 100, minimumSeverity: 'HIGH'
```
But this is the pipeline grammar of warnings-ng, I'm not sure how to override the configurations in `buildPlugin`, maybe 
`buildPlugin(java: minimumSeverity: 'LOW')`? @timja 

Also, do you think we should provide another configuration for minimal severity beside the warnings-ng one to only control the published checks?

Reply:
I don’t think so, just need to configure warnings ng / the Pom appropriately

Reply:
see #154